test/python/transpiler/test_split_2q_unitaries.py

```diff
@@ -377,3 +384,49 @@ def test_2q_swap_with_large_circuit(self):
         )  # the original 2-qubit unitary should be split into 2 1-qubit unitaries.
         self.assertTrue(expected_op.equiv(res_op))
         self.assertTrue(matrix_equal(expected_op.data, res_op.data, ignore_phase=False))
+
+    @combine(
+        swap_list=[
+            [(0, 1), (1, 0)],
+            [(0, 1), (0, 2), (0, 3), (0, 4)],
+            [(0, 3), (5, 6), (2, 3), (1, 5), (2, 1)],
+            [(0, 5), (2, 1), (1, 2), (6, 0), (3, 2), (4, 2), (0, 1), (4, 2), (1, 3), (6, 4)],
+        ],
+        elide_before=[True, False],
+        seed=[42, 13, 55],
+    )
+    def test_2q_swap_with_elide_permutations(self, swap_list, elide_before, seed):
+        """Tests compatability of 2q-swap with the elide permutations pass"""
+        qc = QuantumCircuit(2)
+        qc.swap(0, 1)
+        qc.global_phase += 1.2345
+        np.random.seed(seed)
+        num_qubits = 7
+        qc_split = QuantumCircuit(num_qubits)
+        unitary_swap_list = np.random.choice([True, False], size=len(swap_list))
+        expected_layout = list(range(num_qubits))
+        for swap, unitary_swap in zip(swap_list, unitary_swap_list):
+            expected_layout[swap[0]], expected_layout[swap[1]] = (
+                expected_layout[swap[1]],
+                expected_layout[swap[0]],
+            )
+            if unitary_swap:
+                qc_split.append(UnitaryGate(Operator(qc)), swap)
+            else:
+                qc_split.swap(*swap)
+
+        pm = PassManager()
+        if elide_before:
+            pm.append(ElidePermutations())
+            pm.append(Split2QUnitaries(split_swap=True))
+        else:
+            pm.append(Split2QUnitaries(split_swap=True))
+            pm.append(ElidePermutations())
+
+        res = pm.run(qc_split)
+        self.assertEqual(expected_layout, res.layout.final_index_layout())
+
+        res_op = Operator.from_circuit(res)
+        expected_op = Operator(qc_split)
+        self.assertTrue(expected_op.equiv(res_op))
```

Comment:
why do you check here that they are equivalent (=up to a global phase)?
I think that this equality check should work:
`self.assertEqual(expected_op, res_op)`

Reply:
I think I followed suit with other tests, but if your line works I can switch to it.

Reply:
Indeed, I see that many tests have these two lines:
```
        self.assertTrue(Operator(qc).equiv(qc_split))
        self.assertTrue(
            matrix_equal(Operator(qc).data, Operator(qc_split).data, ignore_phase=False)
        )
```
namely, they check that the operators are the same (up to a global phase) and then also check that the matrices are the same (including the global phase). This does not make sense to me.
I think these two lines can be replaced everywhere by this line that asserts that the matrices are the same:
`self.assertEqual(Operator(qc), Operator(qc_split))`

Reply:
I'm ok with it, we can do it as part of this PR, if the others agree.